client_src/js/system/client-settings.js

```diff
@@ -0,0 +1,145 @@
+    document.getElementById("panelCurrentOpOn").checked = clientSettings.panelView.currentOp;
+    document.getElementById("panelHideOffline").checked = clientSettings.panelView.hideOff;
+    document.getElementById("panelHideClosed").checked = clientSettings.panelView.hideClosed;
+    // document.getElementById("panelHideIdle").checked =
+    //   res.panelView.hideIdle;
+    if (clientSettings.panelView.printerRows) {
+      document.getElementById("selectCameraGrid").value = clientSettings.panelView.printerRows;
+    } else {
+      document.getElementById("selectCameraGrid").value = 2;
+    }
+
+    if (clientSettings.dashboard) {
+      document.getElementById("currentOperations").checked =
+        clientSettings.dashboard.farmActivity.currentOperations;
+      document.getElementById("cumulativeTimes").checked =
+        clientSettings.dashboard.farmActivity.cumulativeTimes;
+      document.getElementById("averageTimes").checked =
+        clientSettings.dashboard.farmActivity.averageTimes;
+
+      document.getElementById("printerState").checked =
+        clientSettings.dashboard.printerStates.printerState;
+      document.getElementById("printerTemps").checked =
+        clientSettings.dashboard.printerStates.printerProgress;
+      document.getElementById("printerUtilisation").checked =
+        clientSettings.dashboard.printerStates.printerUtilisation;
+      document.getElementById("printerProgress").checked =
+        clientSettings.dashboard.printerStates.printerProgress;
+      document.getElementById("currentStatus").checked =
+        clientSettings.dashboard.printerStates.currentStatus;
+
+      document.getElementById("currentUtilisation").checked =
+        clientSettings.dashboard.farmUtilisation.currentUtilisation;
+      document.getElementById("farmUtilisation").checked =
+        clientSettings.dashboard.farmUtilisation.farmUtilisation;
+
+      document.getElementById("weeklyUtilisation").checked =
+        clientSettings.dashboard.historical.weeklyUtilisation;
+      document.getElementById("hourlyTotalTemperatures").checked =
+        clientSettings.dashboard.historical.hourlyTotalTemperatures;
+      document.getElementById("environmentalHistory").checked =
+        clientSettings.dashboard.historical.environmentalHistory;
+      document.getElementById("filamentUsageCheck").checked =
+        clientSettings.dashboard.historical.filamentUsageByDay;
+      document.getElementById("printCompletionCheck").checked =
+        clientSettings.dashboard.historical.historyCompletionByDay;
+      document.getElementById("filamentUsageOverTimeCheck").checked =
+        clientSettings.dashboard.historical.filamentUsageOverTime;
+    }
+
+    if (clientSettings.controlSettings) {
+      document.getElementById("printerControlFilesFirst").checked =
+        clientSettings.controlSettings.filesTop;
+    }
+  }
+
+  static async update() {
+    const opts = {
+      panelView: {
```

Comment:
Lets make a `shared` folder after all this to share code between backend and frontend. I know from experience that Vue and React dont like this due to isolation, but jsDOM dont care xD

Reply:
Yeah would be good